lib/class_relationships.rb

```diff
@@ -0,0 +1,67 @@
+# Think of these toy objects as the singletons
+ToyObject = BasicObject.new
+
+def ToyObject.to_s
+  inspect
+end
+
+def ToyObject.inspect
+  "ToyObject"
+end
+
+def ToyObject.toy_class
+  ToyClass
+end
+
+def ToyObject.toy_kind_of?(klass)
+  superclass = toy_class
+  while superclass
+    return true if klass == superclass
+    superclass = superclass.toy_superclass
```

Comment:
NEAT!! Ruby trivia is my fave ❤️  I got curious about this and was playing around, and realized that we could rewrite `#toy_kind_of` like this:
```ruby
  def ToyObject.toy_kind_of?(klass)
    superclass = toy_class
    Enumerator.produce(superclass) { |klass| superclass = klass.superclass or raise StopIteration }
  end
```
And this even handles cases where our `return true if klass == superclass` line would break (as discussed [here](https://github.com/adrianna-chang-shopify/ruby-object-model/pull/19#discussion_r752079349)), because we raise `StopIteration` as soon as we can no longer do `klass.superclass` (which we can't for `ToyObject`). Cool!

Reply:
NEAT!! Ruby trivia is my fave :heart:  I got curious about this and was playing around, and realized that we could rewrite `#toy_kind_of` like this:   
```ruby
  def ToyObject.toy_kind_of?(klass)
    superclass = toy_class
    e = Enumerator.produce(superclass) do |klass|
      raise StopIteration if klass == ToyObject
      superclass = klass.toy_superclass
    end
    e.include?(klass)
end
```

Reply:
Nice. Can you explain what the reassignment to the `superclass` local variable is doing?

Reply:
I was looking at [these docs](https://rubydoc.info/stdlib/core/Enumerator.produce), and saw that they were reassigning the value to the local variable that was used as the `initial` value in this example:
```ruby
ancestors = Enumerator.produce(node) { |prev| node = prev.parent or raise StopIteration }
```
. In rereading the docs though, I noticed that it says
> The result of the previous iteration is passed to the next one. 

which makes me think reassigning is not necessary at all because `klass.toy_superclass` will just be the result of the block. 😅 I'm wondering if maybe they were reassigning in the example above because the `or raise` portion was following the iteration to the next node?

Reply:
I think I answered my own question here 😂 After changing `ToyObject.toy_superclass` to return `nil`, I tweaked this to be
```ruby

  def ToyObject.toy_kind_of?(klass)
    superclass = toy_class
    e = Enumerator.produce(superclass) { |klass| klass.toy_superclass or raise StopIteration }
    e.include?(klass)
  end
```
and it works, so I think _even_ with adding `or raise StopIteration` there's no need for reassigning here?

Reply:
Yeah, I agree. Assigning the next value to a local variable _happens_ to return it from the block, but it’s not necessary here. Neat!

(I hadn’t noticed that the documentation example had an assignment. That’s not very helpful.)